### PR TITLE
Add four note styles, images max-height, styles for REST api, search result page, "show more/less" code button

### PR DESCRIPTION
**Pull request by dr3amf4ll**
Hot to use notes:
1. Default note:
   > Hi, i'm note
2. Tip:
   > Hi, i'm tip
   {: .tip}
3. Important note:
   > Hi, i'm important note
   {: .important}
4. Caution note:
   > Hi, i'm caution note
   {: .caution}

tips

How REST api page now looks:

rest

How button "show more/less" code now looks:

more_less

How search result page now looks:

search

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

File: gh-pages-stub/public/css/styles.css [hunk 1]
```diff
@@ -467,24 +467,81 @@ img {
...
+}
+blockquote.important table{
+	outline: 2px solid #FFF2DF;
+}
+blockquote.caution{
+	border-left: 5px solid #f22f2f;
+    background: #feeaea;
+}
+blockquote.caution:before{
+	background-image: url(/public/notes/caution.png);
```
Comment (balldir) [inline gh-pages-stub/public/css/styles.css]: There will be an issue with it on github pages.
It should be written as 

> background-image: url({{ site.baseurl }}/public/notes/caution.png);

And the front matter block should be added to the beginning of the file

> +---
> +sitemap: false
> +---

Unfortunately, this change is not compatible with current development branch. I think this pr should be updated after merge of https://github.com/kaaproject/kaa/pull/844

Comment (balldir) [inline gh-pages-stub/public/css/styles.css]: @dr3amf4ll , https://github.com/kaaproject/kaa/pull/844 was just merged.
Please, update patch with next changes 

>  blockquote.important:before{
>   background-image: url({{ site.baseurl }}/public/notes/important.png);
>  }
>  blockquote.important table{
>   outline: 2px solid #FFF2DF;
>  }
>  blockquote.caution{
>   border-left: 5px solid #f22f2f;
>      background: #feeaea;
>  }
>  blockquote.caution:before{
>   background-image: url({{ site.baseurl }}/public/notes/caution.png);

And, please, check why this patch affects menu text align https://drive.google.com/file/d/0B6sIF8dhD0gVRXFMd3VaZTZqdTQ/view?usp=sharing

**Comment by balldir** [comment]
LGTM

**Comment by hunch7** [comment]
LGTM

**Pull request merged**